### PR TITLE
Add rigid property to all items and usages

### PR DESCRIPTION
**Pull request by mugling**
This PR extends the `rigid` property to be supported by all items. For non-rigid worn items encumbrance now scales with increasing volume. Ammo belts, some ammo pouches and holsters plus waterskins are affected.

Currently the penalty is `base + (contained volume * 2)` but following playtesting we may consider changing that constant and/or allowing specification in JSON, possibly as `bulkiness`

Holsters are normalized (9371f46) and some balancing changes made:
- **holster** remains the default choice for a typical sidearm
- **fast draw holster** now also has lower encumbrance to balance the recent nerfs to `draw_speed`. If your sidearm will fit is preferable over the standard holster.
- **ankle holster** now has `max_weight` of 1kg to limit you to a realistic set of holdout pistols
- **XL holster** loses one unit of volume (to become 8) and is now non-rigid so whilst you can still strap a long gun to either leg you might have some difficulties running if you follow this plan. The holster itself has no base encumbrance and remains `LEG_EITHER` so the penalty is effectively halved.
- **back holster** is unchanged. Practice does improve `draw_speed` whereas the shoulder strap does not, however remains a weak item overall. Suggestions?

Ammo pouches are normalized (4fcd4f6) and some balancing changes made:
- **ankle pouch** is intended for pistol magazines and such. Few disadvantages but low volume and competes with the same slot as the ankle sheath and ankle holster.
- **leg pouch** drops a unit of `volume` and becomes non-rigid with zero base `encumbrance`. As a result they only add penalty when filled so most players will want to sew a pair. Fits the majority of magazines and can be considered the default type.
- **chest pouch** is a larger alternative to the leg pouch. Less appealing as covers `TORSO` layer but necessary if your chosen magazines are larger or you need the extra capacity overall. Same encumbrance mechanics as the leg pouch (pay for what you use).
- **chest rig** has slightly lower base `encumbrance` and `storage` but is one of only two items than can contain the `MAG_BULKY` magazines. Unlike the ammo satchel it's ammo pouch is rigid so adding a magazine won't slow you down. Probably the best choice if you have a stash of the uncommon `MAG_BULKY` magazines.
- **ammo satchel** can store magazines of any size including `MAG_BULKY`. It is non-rigid with low base `encumbrance` so is relatively low penalty except for huge magazines in which case it might be worth dropping before melee. The game limits you to a pair. Future extension (separate PR) will add support for `multi` so you can stash several magazines.

### CODE REVIEW

**Comment by kevingranade** [comment]
An observation about ankle holsters, weight on your ankles slows you down
very disproportionately compared to weight on other parts of your body, is
this reflected?

**Comment by mugling** [comment]
No, it's not directly modeled (weight doesn't affect encumbrance), however it is the inspiration for applying `max_weight`.

In the long run (most definitely separate PR) how would you feel about calculating encumbrance as a function of location, coverage, weight, materials and their thickness plus a scaling factor for each item `bulkiness`?

**Comment by Coolthulhu** [comment]
> In the long run (most definitely separate PR) how would you feel about calculating encumbrance as a function of location, coverage, weight, materials and their thickness plus a scaling factor for each item bulkiness?

This shouldn't disable the ability to specify base encumbrance, though.
There could be an extra penalty due to changed state (weight increase due to contents, thickness increase due to reinforcement etc.) and possibly a function to calculate encumbrance for items with unset encumbrance (to unbloat jsons).
Alternatively, the function could be used for all items, but some items would need an extra json field to describe encumbrance deviation from the formula.

**Comment by mugling** [comment]
> This shouldn't disable the ability to specify base encumbrance, though.
> There could be an extra penalty due to changed state

This is the approach this PR takes

> Alternatively, the function could be used for all items, but some items would need an extra json field to describe encumbrance deviation from the formula.

Long-term this is probably the better approach

**Comment by Coolthulhu** [comment]
> fast draw holster now also has lower encumbrance to balance the recent nerfs to draw_speed. If your sidearm will fit is preferable over the standard holster.

Lower than standard one? It shouldn't be 0 encumbrance.

Do holster need those VARSIZE tags? How do you refit a holster?

**Comment by Rivet-the-Zombie** [comment]
>How do you refit a holster?

Our holsters seem to come with fasteners and straps that can be adjusted, since they don't require a separate belt to wear them on.

**Comment by mugling** [comment]
> since they don't require a separate belt to wear them on.

...yet

**Comment by mugling** [comment]
> Do holster need those VARSIZE tags?

No, lets drop that

**Comment by Rivet-the-Zombie** [comment]
`base + (contained volume * 2)` seems like a lot.

**Comment by mugling** [comment]
@Rivet-the-Zombie I've dropped it to `base + contained_volume` in b7c9b3f for the time being although eventually (separate PR) we could go with an algorithm based on other item properties.

**Comment by Rivet-the-Zombie** [comment]
Yeah, it seems like something we may want to tweak after further testing.

**Comment by Rivet-the-Zombie** [comment]
>Future extension (separate PR) will add support for multi so you can stash several magazines.

What about tactical holsters that can store an additional magazine along with a handgun?

File: src/item_factory.cpp [hunk 1]
```diff
@@ -872,7 +872,6 @@ void Item_factory::load( islot_container &slot, JsonObject &jo )
     slot.seals = jo.get_bool( "seals", false );
     slot.watertight = jo.get_bool( "watertight", false );
     slot.preserves = jo.get_bool( "preserves", false );
-    slot.rigid = jo.get_bool( "rigid", false );
```
Comment (BevapDin) [inline src/item_factory.cpp]: Now the "rigid" flag in the container slot data in JSON will be completely ignored. If you change the way the JSON data is read, you should update the JSON (including the contents of mods) to match the new syntax.

- Move the member from the container slot into the common item data (for example needed for the jerrypack), the "rigid" member inside the container slot is not used anymore.

- Fix the containers that don't have an explicit "rigid" value set. The PR changes the default from `false` (for containers) to `true`, so all containers that don't have an explicit value set (and were `rigid = false`) are now `rigid = true`.

Test this with a few containers. I have a "large waterskin of water (20)" that has volume 3, when emptied, it still has volume 3.

**Comment by mugling** [comment]
> What about tactical holsters that can store an additional magazine along with a handgun?

The issue there is the interface

**Comment by mugling** [comment]
@BevapDin good point and fixed in a48ee38 for mainline and for any mods in 376678ea

**Comment by Rivet-the-Zombie** [comment]
>The issue there is the interface

Simple enough to make an interface for that; activate the holster and it gives you three options in a menu - store/draw gun, store/draw magazine, cancel. If it has both inside of it, it displays as `tactical holster with ~gun name~ and mag`.

**Comment by mugling** [comment]
The problem then is that some descriptions can become too long.

The holstering code for multiple knives in boots shows `full` for this reason when you have a knife in both boots as there just isn't enough space.

We could just go with `|| holster (left) with Glock 19 (15) and mag (15)` but even that's quite lengthy or the plausibly worse example `|| ankle holster (left) with SIG Sauer P230 (10) and mag (10)`.

Eventually I'm considering adding a `Holsters` section below `Worn` which shows items you have immediately to hand.

**Comment by macrosblackd** [comment]
Why not list the holstered items beneath the worn item, indented and maybe a lighter color or smaller font?

**Comment by mugling** [comment]
Special-casing the UI code for this would be painful. In run a nested list might be the best way forward

**Comment by Coolthulhu** [comment]
Moderately-small problem:
Fast draw holster is significantly less encumbering than regular holster AND has the advantage of fast draw. The only reason to use regular holster over fast draw is that the gun is exactly 5 volume.

**Comment by mugling** [comment]
> Fast draw holster is significantly less encumbering than regular holster AND has the advantage of fast draw. The only reason to use regular holster over fast draw is that the gun is exactly 5 volume.

It might be worth dropping fast-draw to volume 3

**Pull request merged**